ssl/quic/qlog_event_helpers.c

```diff
@@ -0,0 +1,627 @@
+
+    case QUIC_CHANNEL_STATE_ACTIVE:
+        if (handshake_confirmed)
+            return "handshake_confirmed";
+        else if (handshake_complete)
+            return "handshake_complete";
+        else
+            return "attempted";
+
+    case QUIC_CHANNEL_STATE_TERMINATING_CLOSING:
+        return "closing";
+
+    case QUIC_CHANNEL_STATE_TERMINATING_DRAINING:
+        return "draining";
+
+    case QUIC_CHANNEL_STATE_TERMINATED:
+        return "closed";
+    }
+}
+#endif
+
+void ossl_qlog_event_connectivity_connection_state_updated(QLOG *qlog,
+                                                           uint32_t old_state,
+                                                           uint32_t new_state,
+                                                           int handshake_complete,
+                                                           int handshake_confirmed)
+{
+#ifndef OPENSSL_NO_QLOG
+    const char *state_s;
+
+    QLOG_EVENT_BEGIN(qlog, connectivity, connection_state_updated)
+        state_s = map_state_to_qlog(new_state,
+                                    handshake_complete,
+                                    handshake_confirmed);
+
+        if (state_s != NULL)
+            QLOG_STR("state", state_s);
+    QLOG_EVENT_END()
+#endif
+}
+
+#ifndef OPENSSL_NO_QLOG
+static const char *quic_err_to_qlog(uint64_t error_code)
+{
+    switch (error_code) {
+        case QUIC_ERR_INTERNAL_ERROR:
+            return "internal_error";
+        case QUIC_ERR_CONNECTION_REFUSED:
+            return "connection_refused";
+        case QUIC_ERR_FLOW_CONTROL_ERROR:
+            return "flow_control_error";
+        case QUIC_ERR_STREAM_LIMIT_ERROR:
+            return "stream_limit_error";
+        case QUIC_ERR_STREAM_STATE_ERROR:
+            return "stream_state_error";
+        case QUIC_ERR_FINAL_SIZE_ERROR:
+            return "final_size_error";
+        case QUIC_ERR_FRAME_ENCODING_ERROR:
+            return "frame_encoding_error";
+        case QUIC_ERR_TRANSPORT_PARAMETER_ERROR:
+            return "transport_parameter_error";
+        case QUIC_ERR_CONNECTION_ID_LIMIT_ERROR:
+            return "connection_id_limit_error";
+        case QUIC_ERR_PROTOCOL_VIOLATION:
+            return "protocol_violation";
+        case QUIC_ERR_INVALID_TOKEN:
+            return "invalid_token";
+        case QUIC_ERR_APPLICATION_ERROR:
+            return "application_error";
+        case QUIC_ERR_CRYPTO_BUFFER_EXCEEDED:
+            return "crypto_buffer_exceeded";
+        case QUIC_ERR_KEY_UPDATE_ERROR:
+            return "key_update_error";
+        case QUIC_ERR_AEAD_LIMIT_REACHED:
+            return "aead_limit_reached";
+        case QUIC_ERR_NO_VIABLE_PATH:
+            return "no_viable_path";
+        default:
+            return NULL;
+    }
+}
+#endif
+
+void ossl_qlog_event_connectivity_connection_closed(QLOG *qlog,
+                                                    const QUIC_TERMINATE_CAUSE *tcause)
```

Comment:
Since our style has us indending to a parenthesis, it is not really possible to shorten this without changing the function name. For this reason it can't really be enforced as a strict rule and it's left to judgement.